### PR TITLE
Add 0.12.0 release notes pt 2

### PR DESCRIPTION
**Pull request by cwsteinbach**

### CODE REVIEW

File: site/docs/releases.md
Comment (jackye1995) [inline site/docs/releases.md:72]: Configuring catalog using `iceberg.mr.catalog` is now deprecated. (https://github.com/apache/iceberg/pull/2565)

Multi table insert support is added (no multi-table ACID transaction guarantee) (https://github.com/apache/iceberg/pull/2228)

**Comment by jackye1995** [comment]
Thanks for the additional patch! I read through a few Hive related classes and those are all I could find. @marton-bod  @lcspinter @pvary anything else on your side that should be noted? I also saw a few bug fixes done by Marton for Tez issues.

**Comment by marton-bod** [comment]
Thanks @jackye1995! I think we might not want to list the Tez fixes, because those were write-related fixes and we don't yet support Tez writes - so don't want to confuse users :) 

However, it might be worth including these:
Hive: enable dropping HMS tables despite metadata problems (#2583)
Hive: Add table-level JVM lock on commits (#2547) 

But it's up to you, not sure what patches make the cut into the release docs. Thanks again for the call-out!

**Comment by marton-bod** [comment]
One more new feature to consider: 
Hive: Vectorized ORC reads for Hive #2613

**Comment by cwsteinbach** [comment]
@jackye1995 @marton-bod I added the following PRs:
#2565, #2583, #2547, #2613.

@jackye1995 please take a look and approve if everything looks good. Thanks!

File: site/docs/releases.md
Comment (jackye1995) [inline site/docs/releases.md:75]: nit: 2 "deprecated". I think we should say "Configuring Iceberg catalog implementation in Hive using `iceberg.mr.catalog` is now deprecated"

File: site/docs/releases.md
Comment (jackye1995) [inline site/docs/releases.md:81]: `REPLACE PARTITION FIELD`, not `CHANGE PARTITION FIELD`

Comment (jackye1995) [reply site/docs/releases.md:81]: And I don't think we need the full SQL sentence, just `ALTER TABLE REPLACE PARTITION FIELD` should be good.

File: site/docs/releases.md
Comment (cwsteinbach) [reply site/docs/releases.md]: Done.

File: site/docs/releases.md
Comment (cwsteinbach) [reply site/docs/releases.md]: Done

**Review by jackye1995 — APPROVED** [review]
thank you for the changes! @danielcweeks any other comments on your side?

**Comment by rdblue** [comment]
I'm going to spend some time today going through commits. I got a start at it yesterday and found several things missing, including some that were caught here, so I'd like to make sure that we get it all.

File: site/docs/releases.md
Comment (danielcweeks) [inline site/docs/releases.md:85]: Spark: Fix nested struct pruning (#2877)

File: site/docs/releases.md
Comment (danielcweeks) [inline site/docs/releases.md:95]: nit: missing 'k' in spark

File: site/docs/releases.md
Comment (danielcweeks) [inline site/docs/releases.md:70]: Core: Support nulls in StructLike collections (#2929)

Comment (danielcweeks) [reply site/docs/releases.md:70]: Core: Fix partition field IDs in table replacement (#2906)

Comment (danielcweeks) [reply site/docs/releases.md:70]: Core: Add check-status config to TableProperties (#2596)

Note: This probably needs more context since it's intended to help address issues with Metastore timeouts, but is not clear from the commit message.

File: site/docs/releases.md
Comment (danielcweeks) [reply site/docs/releases.md:74]: Hive: Vectorized ORC reads for Hive (#2613)

File: site/docs/releases.md
Comment (rdblue) [reply site/docs/releases.md:70]: Yeah, that's another one I flagged. The original commit is #2328 where we introduced a new exception to signal that commit state was uncertain and status checks. There were a few follow-ups to that as well, including this one that adds configuration for the checks, and support for the exception in Glue (#2402) and Nessie (#2515).

I think this deserves a high-level callout because it is a big fix that a lot of people are waiting on. I think Shopify was waiting for it, for example.

Comment (rdblue) [reply site/docs/releases.md:70]: I'm not sure I'd include #2929 since that was extending an internal class for another use rather than fixing a user-facing problem. Up to you though.

File: site/docs/releases.md
Comment (rdblue) [inline site/docs/releases.md:96]: Isn't this a new feature? I would list it in improvements, not bug fixes.

File: site/docs/releases.md
Comment (rdblue) [inline site/docs/releases.md:92]: I don't think this is notable. This just passes defaults.

File: site/docs/releases.md
Comment (rdblue) [inline site/docs/releases.md:91]: I think a better description would be "Fixed Parquet dictionary filtering with fixed-length byte arrays and decimals".

File: site/docs/releases.md
Comment (rdblue) [inline site/docs/releases.md:78]: "Added support for"?

File: site/docs/releases.md
Comment (rdblue) [inline site/docs/releases.md:77]: This is a bug fix so it should probably go in bugs. If possible, can you mention cases where one may have hit this issue?

File: site/docs/releases.md
Comment (rdblue) [inline site/docs/releases.md:75]: May want to mention that this is a result of the above configuration change.

File: site/docs/releases.md
Comment (rdblue) [inline site/docs/releases.md:69]: Isn't this a Spark-specific change?

File: site/docs/releases.md
Comment (rdblue) [inline site/docs/releases.md:80]: I think it would be better to replace "Added extensions DDL" with "Added SET and DROP IDENTIFIER FIELDS to ALTER TABLE"

File: site/docs/releases.md
Comment (rdblue) [inline site/docs/releases.md:89]: In addition, #2308 was fixed by #2865. This added a validation that files referenced by row-level deletes are not concurrently rewritten.

File: site/docs/releases.md
Comment (rdblue) [inline site/docs/releases.md:84]: A few more for Spark:
* Spark 3.1 is now supported (#2512)
* Added RemoveReachableFiles action (#2415)
* Added add_files stored procedure (#2210)
* Refactored Actions API and added a new entry point (need to find the ref)
* Added support for Hadoop configuration overrides (#2922)

File: site/docs/releases.md
Comment (rdblue) [inline site/docs/releases.md:94]: A couple more Spark additions:
* Fixed NaN handling for float and double metrics (#2464)
* Fixed Kryo serialization for data and delete files (#2343)

File: site/docs/releases.md
Comment (rdblue) [inline site/docs/releases.md:89]: Another general one: Parquet: Fix row group filters with promoted types (#2232)

File: site/docs/releases.md
Comment (rdblue) [reply site/docs/releases.md:70]: Also: Added key_metadata to manifest lists for encryption (#2675)

Comment (cwsteinbach) [reply site/docs/releases.md:70]: @danielcweeks, can you please suggest a blurb for @2596?

File: site/docs/releases.md
Comment (cwsteinbach) [reply site/docs/releases.md:69]: The labels on #1870 indicate otherwise. I also don't see anything in the code that would flag this as something that only applies to Spark. If I'm wrong please explain what I'm missing.

File: site/docs/releases.md
Comment (cwsteinbach) [reply site/docs/releases.md:89]: Since you fixed #2865 can you please provide a blurb for it?

File: site/docs/releases.md
Comment (cwsteinbach) [reply site/docs/releases.md:77]: @rdblue are you referring to #2583?

File: site/docs/releases.md
Comment (cwsteinbach) [reply site/docs/releases.md:75]: @danielcweeks Hive vectorized reads were already mentioned in the "Other notable changes" section. I moved them to the high-level changes section.

File: site/docs/releases.md
Comment (cwsteinbach) [reply site/docs/releases.md]: Fixed.

File: site/docs/releases.md
Comment (cwsteinbach) [reply site/docs/releases.md]: Fixed.

File: site/docs/releases.md
Comment (cwsteinbach) [reply site/docs/releases.md:99]: Fixed.

File: site/docs/releases.md
Comment (cwsteinbach) [reply site/docs/releases.md]: Fixed.

File: site/docs/releases.md
Comment (cwsteinbach) [reply site/docs/releases.md]: Fixed.

File: site/docs/releases.md
Comment (cwsteinbach) [reply site/docs/releases.md]: Fixed.

File: site/docs/releases.md
Comment (cwsteinbach) [reply site/docs/releases.md]: Fixed.

File: site/docs/releases.md
Comment (cwsteinbach) [reply site/docs/releases.md:85]: Fixed.

File: site/docs/releases.md
Comment (cwsteinbach) [reply site/docs/releases.md]: Fixed.

File: site/docs/releases.md
Comment (cwsteinbach) [reply site/docs/releases.md:96]: Fixed.

File: site/docs/releases.md
Comment (cwsteinbach) [reply site/docs/releases.md:70]: I added blurbs for #2929, #2906, #2675. Waiting on @danielcweeks for a blurb about #2596.

File: site/docs/releases.md
Comment (cwsteinbach) [reply site/docs/releases.md:78]: >  If possible, can you mention cases where one may have hit this issue?

@pvary and @marton-bod: Please provide the examples requested by @rdblue. Thanks.

Comment (rdblue) [reply site/docs/releases.md:78]: Nevermind then. Let's not block on this.

Comment (marton-bod) [reply site/docs/releases.md:78]: The fix was following up on the issue: https://github.com/apache/iceberg/issues/2554. 
If the table becomes unloadable for any reason (e.g. the metadata files have been deleted/corrupted on the FS by some outside process), afterwards you were not able to ever drop your table from HMS before this fix

File: site/docs/releases.md
Comment (rdblue) [reply site/docs/releases.md:96]: > Added validation that files referenced by row-level deletes are not concurrently rewritten (#2308)

Comment (cwsteinbach) [reply site/docs/releases.md:96]: Done.

File: site/docs/releases.md
Comment (cwsteinbach) [reply site/docs/releases.md:78]: Added. Done.

**Pull request merged**
